### PR TITLE
Adding nxos_static_route

### PR DESCRIPTION
**Pull request by GGabriele**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

nxos_static_route
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/']
```
##### SUMMARY

Manages static route configuration

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @GGabriele. This module is going into community review.

We notice that you are one of the maintainers (@privateip, @GGabriele, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/nxos/nxos_static_route.py [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+author: Gabriele Gerbino (@GGabriele)
+extends_documentation_fragment: nxos
+notes:
+    - If no vrf is supplied, vrf is set to default
+    - If I(state)=absent, the route will be removed, regardless of the
+      non-required parameters.
+options:
+    prefix:
+        description:
+            - Destination prefix of static route
```
Comment (gundalow) [inline network/nxos/nxos_static_route.py]: Please add a fullstop.

File: network/nxos/nxos_static_route.py [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+        required: false
+        default: null
+    route_name:
+        description:
+            - Name of the route. Used with the name parameter on the CLI.
+        required: false
+        default: null
+    pref:
+        description:
+            - Preference or administrative difference of route (range 1-255)
```
Comment (gundalow) [inline network/nxos/nxos_static_route.py]: Please add a fullstop.

File: network/nxos/nxos_static_route.py [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+        required: false
+        default: null
+    state:
+        description:
+            - Manage the state of the resource
```
Comment (gundalow) [inline network/nxos/nxos_static_route.py]: Please add a fullstop.

File: network/nxos/nxos_static_route.py [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+module: nxos_static_route
+version_added: "2.2"
+short_description: Manages static route configuration
+description:
+    - Manages static route configuration
+author: Gabriele Gerbino (@GGabriele)
+extends_documentation_fragment: nxos
+notes:
+    - If no vrf is supplied, vrf is set to default
+    - If I(state)=absent, the route will be removed, regardless of the
```
Comment (gundalow) [inline network/nxos/nxos_static_route.py]: `C(state=absent)`

File: network/nxos/nxos_static_route.py [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+            - Destination prefix of static route
+        required: true
+    next_hop:
+        description:
+            - Next hop address or interface of static route.
+              If interface, it must be the fully-qualified interface name.
+        required: true
+    vrf:
+        description:
+            - VRF for static route
```
Comment (gundalow) [inline network/nxos/nxos_static_route.py]: Please add a fullstop.

**Comment by gundalow** [comment]
needs_revision
@jedelman8 Could you please cast your eye over this, thanks

**Comment by gregdek** [comment]
Thanks @GGabriele for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**